src/stdlib/parse_float.rs

```diff
@@ -0,0 +1,142 @@
+use crate::compiler::prelude::*;
+use std::str::FromStr;
+
+fn parse_float(value: Value) -> Resolved {
+    let string = value.try_bytes_utf8_lossy()?;
+    let float = f64::from_str(&string)
+        .map_err(|err| format!("could not parse float: {err}"))?;
+    let converted = NotNan::new(float)
+        .map_err(|_| "NaN number not supported")?;
+
+    Ok(converted.into())
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseFloat;
+
+impl Function for ParseFloat {
+    fn identifier(&self) -> &'static str {
+        "parse_float"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+        ]
```

Comment:
To make this function a bit more generic, let's add a base parameter like we do in [parse_int](https://github.com/vectordotdev/vrl/blob/main/src/stdlib/parse_int.rs#L49).

Reply:
Since the Rust standard library doesn't support parsing float with other radix. To support this, we have to implement the whole parsing logic by ourselves. Also, I think it is very rare to see float numbers in other radix. In my opinion, it is not worthy to support the base parameter.

Reply:
This makes sense, I agree. We can go ahead with this PR as is. Note, there are some failing checks.